File: _static/code/harrison_kreps/hk_price_singlebeliefs.py
"""

Authors: Chase Coleman, Tom Sargent

"""
import scipy.linalg as la

def price_single_beliefs(transition, dividend_payoff, β=.75):
    """
    Function to Solve Single Beliefs
    """
    # First compute inverse piece
    imbq_inv = la.inv(np.eye(transition.shape[0]) - β * transition)

    # Next compute prices
    prices = β * imbq_inv @ transition @ dividend_payoff

    return prices

File: rst_files/_static/code/harrison_kreps/hk_price_singlebeliefs.jl
#=

Authors: Shunsuke Hori

=#
"""
Function to Solve Single Beliefs
"""
function price_single_beliefs(transition::Matrix, dividend_payoff::Vector;
                              β::AbstractFloat=.75)
    # First compute inverse piece
    imbq_inv = inv(eye(size(transition, 1)) - β * transition)

    # Next compute prices
    prices = β * ((imbq_inv * transition) * dividend_payoff)

    return prices
end
